src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const fromParts = fromFormat.slice(0, -1);
+  const toParts = toFormat.slice(0, -1);
+
+  const dateParts = date.split(fromSeparator);
```

Comment:
Consider adding validation to ensure that the 'date' string splits correctly into the expected number of parts. If the number of parts does not match the length of 'fromParts', the function may behave unexpectedly.